src/edit_item.py

```diff
@@ -30,9 +30,9 @@ def __init__(self, application, item):
         utilities.populate_chooser(self.markers, sorted(misc.MARKERS.keys()))
         self.load_values()
         utilities.populate_chooser(self.item_selector, names)
-
-        self.item_selector.set_selected(names.index(self.item.name))
+        self.item_selector.set_selected(names.index(item.name))
         self.set_transient_for(self.props.application.main_window)
+        self.connect("close-request", self.on_close)
```

Comment:
I actually thought I had already removed this line... but turns out I didn't solve the underlying issue yet.

This works as a workaround, but at least at one point we should solve it in the blueprint itself. For some reason the close-request signal does not connect properly to this function from the blueprint.